redfish-core/lib/log_services.hpp

```diff
@@ -2595,6 +2595,122 @@ inline void requestRoutesDBusCELogEntry(App& app)
             });
 }
 
+inline void requestRoutesDBusEventLogEntryDownloadPelJson(App& app)
+{
+    BMCWEB_ROUTE(app, "/redfish/v1/Systems/system/LogServices/EventLog/Entries/"
+                      "<str>/OemPelAttachment")
+        .privileges(redfish::privileges::getLogEntry)
+        .methods(boost::beast::http::verb::get)(
+            [](const crow::Request&,
+               const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+               const std::string& param)
+
+            {
+                std::string entryID = param;
+                dbus::utility::escapePathForDbus(entryID);
+
+                auto eventLogAttachmentCallback = [asyncResp, entryID](
+                                                      bool hiddenPropVal) {
+                    if (hiddenPropVal)
+                    {
+                        messages::resourceNotFound(asyncResp->res, "LogEntry",
+                                                   entryID);
+                        return;
+                    }
+                    auto respHandler = [asyncResp, entryID](
+                                           const boost::system::error_code ec,
+                                           const std::string& pelJson) {
+                        if (ec.value() == EBADR)
+                        {
+                            messages::resourceNotFound(
+                                asyncResp->res, "OemPelLogJsonAttachment",
+                                entryID);
+                            return;
+                        }
+                        if (ec)
+                        {
+                            BMCWEB_LOG_DEBUG << "DBUS response error " << ec;
+                            messages::internalError(asyncResp->res);
+                            return;
+                        }
+
+                        asyncResp->res.jsonValue["Oem"]["IBM"]["PelJson"] =
+                            pelJson;
+                        asyncResp->res.jsonValue["Oem"]["@odata.type"] =
+                            "#OemLogEntryAttachment.Oem";
+                        asyncResp->res.jsonValue["Oem"]["IBM"]["@odata.type"] =
+                            "#OemLogEntryAttachment.IBM";
+                    };
+
+                    crow::connections::systemBus->async_method_call(
+                        respHandler, "xyz.openbmc_project.Logging",
+                        "/xyz/openbmc_project/logging",
+                        "org.open_power.Logging.PEL", "GetPELJSON",
+                        uint32_t(std::stoi(entryID)));
```

Comment:
what if this throw error? 
and it is good to use from_chars.

Reply:
done.